internal/services/containers/kubernetes_cluster_node_pool_resource_test.go

```diff
@@ -1102,6 +1231,128 @@ func TestAccKubernetesClusterNodePool_virtualNetworkOwnershipRaceCondition(t *te
 	})
 }
 
+func TestAccKubernetesClusterNodePool_updateVmSizeAfterFailureWithTempAndOriginal(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_cluster_node_pool", "test")
+	r := KubernetesClusterNodePoolResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.manualScaleVMSkuConfig(data, "Standard_F2s_v2"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That("azurerm_kubernetes_cluster_node_pool.test").ExistsInAzure(r),
+				// create the temporary node pool to simulate the case where both old node pool and temp node pool exist
+				data.CheckWithClientForResource(func(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) error {
+					if _, ok := ctx.Deadline(); !ok {
+						var cancel context.CancelFunc
+						ctx, cancel = context.WithTimeout(ctx, 1*time.Hour)
+						defer cancel()
+					}
+
+					client := clients.Containers.AgentPoolsClient
+
+					id, err := commonids.ParseKubernetesClusterID(state.Attributes["kubernetes_cluster_id"])
+					if err != nil {
+						return err
+					}
```

Comment:
I don't think we need the cluster ID for anything here 
```suggestion
```

Reply:
Removed!